src/main/java/io/rocketpartners/cloud/action/sql/SqlQuery.java

```diff
@@ -244,8 +243,16 @@ else if (term.getToken().indexOf(".") < 0)
       {
          for (Column col : table.getPrimaryIndex().getColumns())
          {
-            Sort sort = new Sort(col.getName(), true);
-            sorts.add(sort);
+            if (parts.select.indexOf('*') >= 0 || parts.select.contains(col.getName()))
+            {
+               Sort sort = new Sort(col.getName(), true);
+               sorts.add(sort);
+            }
+            else
+            {
+               sorts.clear();
+               break;
+            }
```

Comment:
only add the primary index columns for sorting if we're selecting them to begin with.